sdk/python/tests/unit/test_feature_server.py

```diff
@@ -0,0 +1,47 @@
+import json
+from unittest.mock import AsyncMock, MagicMock, patch
+
+import pytest
+from fastapi.testclient import TestClient
+
+from feast import FeatureStore
+from feast.data_source import PushMode
+from feast.feature_server import get_app
+from feast.utils import _utc_now
+from tests.foo_provider import FooProvider
+
+
+@pytest.mark.parametrize(
+    "online_write,push_mode,async_count",
+    [
+        (True, PushMode.ONLINE_AND_OFFLINE, 1),
+        (True, PushMode.OFFLINE, 0),
+        (True, PushMode.ONLINE, 1),
+        (False, PushMode.ONLINE_AND_OFFLINE, 0),
+        (False, PushMode.OFFLINE, 0),
+        (False, PushMode.ONLINE, 0),
+    ],
+)
```

Comment:
test ensures that the only time we attempt an `async store.push` is if online async write is supported and pushmode includes online